src/Resources/sql/segmentAssignment/storedFunctionObject.sql

```diff
@@ -1,24 +1,24 @@
-DROP FUNCTION IF EXISTS PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS;
-DELIMITER //
-
-CREATE FUNCTION PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS(elementIdent INT) RETURNS TEXT
-  READS SQL DATA
-  BEGIN
-    DECLARE segmentIds TEXT;
-    DECLARE elementExists TINYINT;
-    DECLARE breaks TINYINT;
-
-    SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO segmentIds;
-    SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO elementExists;
-    SELECT `breaksInheritance` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO breaks;
-
-    WHILE (elementExists = 0 OR breaks IS NULL OR breaks <> 1) AND elementIdent > 1 DO
-      SELECT `o_parentId`  FROM `objects` WHERE `o_id` = elementIdent INTO elementIdent;
-      SELECT CONCAT_WS(',', segmentIds, (SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object')) INTO segmentIds;
-      SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO elementExists;
-      SELECT `breaksInheritance` INTO breaks FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object';
-    END WHILE;
-
-    RETURN segmentIds;
-  END//
+DROP FUNCTION IF EXISTS PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS;
+DELIMITER //
+
+CREATE FUNCTION PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS(elementIdent INT) RETURNS TEXT
+  READS SQL DATA
+  BEGIN
+    DECLARE segmentIds TEXT;
+    DECLARE elementExists TINYINT;
+    DECLARE breaks TINYINT;
+
+    SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO segmentIds;
+    SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO elementExists;
+    SELECT `breaksInheritance` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO breaks;
+
+    WHILE (elementExists = 0 OR breaks IS NULL OR breaks <> 1) AND elementIdent > 1 DO
+      SELECT `parentId`  FROM `objects` WHERE `id` = elementIdent INTO elementIdent;
```

Comment:
@dvesh3 what about here?

Reply:
I think we have to check column here and run command specifically. please see suggestions here https://stackoverflow.com/questions/133031/how-to-check-if-a-column-exists-in-a-sql-server-table

Reply:
not sure though if this is the best solution